sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -72,7 +74,7 @@ public IReadOnlyList<GeneratorFunctionMetadata> GetFunctionMetadataInfo(List<Met
                         newFunction.IsHttpTrigger = true;
                     }
 
-                    newFunction.RawBindings = bindings;
+                    newFunction.RawBindings = bindings!; // won't be null b/c TryGetBindings would've failed and this line wouldn't be reached
```

Comment:
Oh there are actually some attributes you can apply to give the compiler hints on this: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis, specifically the `NotNullWhenAttribute` here. So you could have:

``` CSharp
bool TryGetBindings(..., [NotNullWhen(true)] out IList<IDictionary<string, string>>? bindings, ...)
```

It means when this function returns `true`, that out param will not be null.